logging/log.go

```diff
@@ -0,0 +1,20 @@
+package logging
+
+import (
+	"context"
+
+	"github.com/spacemeshos/smutil/log"
+)
+
+type LoggerKey struct{}
+
+func NewContext(ctx context.Context, logger log.Log) context.Context {
+	return context.WithValue(ctx, LoggerKey{}, logger)
+}
+
+func FromContext(ctx context.Context) log.Log {
+	if logger, ok := ctx.Value(LoggerKey{}).(log.Log); ok {
+		return logger
+	}
+	return log.AppLog
```

Comment:
I'd rather not fallback to `nil` because all users will have to check against `nil` and fall back to something, risking a segfault. That something will probably be a default logger anyway, so I return it already here. An acceptable, but IMO less useful alternative for our use case is to return a _noop_ logger. The OpenTelemetry go lib does exactly that: https://github.com/open-telemetry/opentelemetry-go/blob/2fe8861a24e20088c065b116089862caf9e3cd8b/trace/context.go#L48-L56